aggregators/1.0/modules/ROOT/pages/aggregators-troubleshooting.adoc

```diff
@@ -0,0 +1,112 @@
+
+To ensure that you use the latest patches, update the Aggregator module to the latest version. In the Apache Maven `pom.xml` file generated by Anypoint Studio replace `x.x.x` with the version that corresponds to the latest version:
+
+[source,xml,linenums]
+----
+<dependency>
+  <groupId>org.mule.modules</groupId>
+  <artifactId>mule-aggregators-module</artifactId>
+  <version>x.x.x</version>
+  <classifier>mule-plugin</classifier>
+</dependency>
+----
+
+To obtain the most up-to-date `pom.xml` file information:
+
+. Go to https://www.mulesoft.com/exchange/[Anypoint Exchange].
+. In Exchange, click *Login* and supply your Anypoint Platform username and password.
+. In Exchange, search for `aggregators`.
+. Select the connector.
+. Click *Dependency Snippets* near the upper right of the screen.
+
+
+== Enable Verbose Logging
+
+To begin troubleshooting Aggregator module, enable verbose logging to see the exact error messages:
+
+. Access Anypoint Studio and navigate to the *Package Explorer* view.
+. Open your application's project name.
+. Open the `src/main/resources` path folder.
+. Open the `log4j2.xml` file inside the folder.
+. If the following line is already in the `log4j2.xml` file, uncomment it to enable it; otherwise, add it:
++
+[source,xml,linenums]
+----
+<AsyncLogger name="org.mule.extension.aggregator" level="DEBUG"/>
+----
++
+[start=6]
+. Save your changes.
+. Click the project name in *Package Explorer* and then click *Run* > *Run As* > *Mule Application*.
+. Navigate to the *Console* view to read the messages in the logger.
+
+
+== Understand the Timeout Behavior in the Size-Based Aggregator
+
+For Size based Aggregator, timeout is the maximum timeout that the aggregation is expected to complete. When the timeout is reached, it does not mean that it is considered complete, but that the waiting time has passed and the aggregation is reset.
+
+For the *Size based aggregator*, the *Timeout* field indicates the maximum time to wait for the aggregation to complete. If the timeout is reached before the total number of elements is equal to the value set in the *Max size* field, the aggregation is not considered complete. The wait time passed and the aggregation is reset. Take into account that you can use an *Aggregator listener* source to capture the aggregated values when giving timeout.
+
+== Make Sure You Are Not in a Multiple Worker Setting
+
+Do not use aggregators with multiple workers and a persistent Object Store because clustering is not available in CloudHub. For details on how workers can be shared or doubled to scale your application and provide high availability, see the xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric]documentation.
+
+== Understand Common Throws
+
+Here is a list of common throw messages and how to interpret them:
+
+For *Size based aggregator*:
+
+* AGGREGATORS:AGGREGATOR_CONFIG
+
+  The Max size or Timeout fields have invalid values, for example, Max size < 0.
```

Comment:
```suggestion
  The *Max size* or *Timeout* fields have invalid values, for example, *Max size* is less than zero.
```

Reply:
...or "...*Max size* is `<0`, depending on whether you are referring to the syntax or the meaning.